boards: nrf52840dongle_nrf52840 is missing storage partition definition

**Describe the bug**

The `storage` partition in `stock` layout (`fstab-stock.dts`) of the `nrf52840dongle_nrf52840` board was removed in 9a842588fb05ee16c8e0a5f7c3217bdfb561c1d4, apparently by a mistake.

**To Reproduce**
Build any sample which makes use of `storage` partition, e.g. OpenThread Co-Processor (RCP config):

`west build -b nrf52840dongle_nrf52840 samples/net/openthread/coprocessor -- -DCONF_FILE="prj.conf overlay-rcp.conf" -DDTC_OVERLAY_FILE="usb.overlay" -DOVERLAY_CONFIG="overlay-usb-nrf-br.conf"`

This will fail with:

`...zephyr/include/zephyr/devicetree.h:190:36: error: 'DT_N_NODELABEL_storage_partition_PARTITION_ID' undeclared (first use in this function)
  190 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)`

**Expected behavior**
Build should pass without any errors.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Version: main

**Additional context**
PR with fix already posted, this bug report is only for formal purposes.

Cheers,
Piotr

--- a/boards/arm/nrf52840dongle_nrf52840/fstab-stock.dts
+++ b/boards/arm/nrf52840dongle_nrf52840/fstab-stock.dts
@@ -23,11 +23,15 @@
 
 		slot0_partition: partition@10000 {
 			label = "image-0";
-			reg = <0x00010000 0x00068000>;
+			reg = <0x00010000 0x00066000>;
 		};
-		slot1_partition: partition@78000 {
+		slot1_partition: partition@76000 {
 			label = "image-1";
-			reg = <0x00078000 0x00068000>;
+			reg = <0x00076000 0x00066000>;
+		};
+		storage_partition: partition@dc000 {
+			label = "storage";
+			reg = <0x000dc000 0x00004000>;
 		};
 
 		/* Nordic nRF5 bootloader <0xe0000 0x1c000>

